File: backend/app/models/dinner.py
from pydantic import BaseModel
from typing import Optional, List
from enum import Enum

class Category(str, Enum):
    MAIN_COURSE = "main_course"
    SOUP = "soup"
    SIDE_DISH = "side_dish"
    SALAD = "salad"
    DESSERT = "dessert"

class Dish(BaseModel):
    id: int
    name: str
    category: Category
    description: Optional[str] = None
    cooking_time: Optional[str] = None
    difficulty: Optional[str] = None
    cuisine_type: Optional[str] = None

class DinnerSuggestion(BaseModel):
    main_course: Optional[Dish] = None
    soup: Optional[Dish] = None
    side_dish: Optional[Dish] = None
    salad: Optional[Dish] = None
    dessert: Optional[Dish] = None

File: frontend/src/types/dinner.ts
export enum Category {
  MAIN_COURSE = "main_course",
  SOUP = "soup",
  SIDE_DISH = "side_dish",
  SALAD = "salad",
  DESSERT = "dessert"
}

export interface Dish {
  id: number;
  name: string;
  category: Category;
  description?: string;
  cooking_time?: string;
  difficulty?: string;
  ingredients?: string[];
}

export interface DinnerSuggestion {
  main_course: Dish;
  soup?: Dish;
  side_dish?: Dish;
  salad?: Dish;
  dessert?: Dish;
}

export interface CategoryResponse {
  category: Category;
  dishes: Dish[];
}

export interface RandomDishResponse {
  dish: Dish;
  category: Category;
}

export interface CategoryOption {
  value: Category;
  label: string;
  color: string;
  icon: string;
}
